src/server/pkg/storage/chunk/chunk.proto

```diff
@@ -17,4 +13,16 @@ message DataRef {
   // The offset and size used for accessing the data within the chunk.
   int64 offset_bytes = 3;
   int64 size_bytes = 4;
+  repeated Tag tags = 5;
+}
+
+message Chunk {
+  string hash = 1;
+  int64 size_bytes = 2;
+  bool edge = 3;
```

Comment:
These fields should be in a different message, or the messages should be renamed. Noun messages such as `Repo`, `Commit`, `Chunk` and `Tag` contain just the information to identify the object, and nothing more. Maybe these could be called `ChunkInfo` and `TagInfo`?

Reply:
Okay, that makes sense. You essentially just want a little more organization of the data stored in the proto. I think this approach is a shoe-in for chunks, but the performance cost could be meaningful for tags. Tags need to be as lightweight as possible and I don't think it will ever make sense to have anything more than an id and size.

Reply:
I don't understand what the performance costs are for `Tag`s, changing the name to `TagInfo` doesn't affect performance at all afaik.

Reply:
I was thinking that you wanted a structure similar to `CommitInfo`:
```
message Tag {
  string id = 1;
}
message TagInfo {
  Tag tag = 1;
  int64 size_bytes = 2;
}
```
There is a performance cost with nested protos that is negligible in most cases. I think it might not be so negligible in the case of tags. If you were thinking just a rename from `Tag` -> `TagInfo`, then I agree there is no performance hit.

Reply:
That was what I was thinking of but I agree it's worth avoiding the overhead for tags, looks like it's 2 bytes nesting a proto. In that case I'd just stick with the name `Tag` since it seems weird to have `TagInfo` but no `Tag`.